assets/js/modules/reader-revenue-manager/datastore/publications.test.js

```diff
@@ -16,115 +17,202 @@
 	);
 
+	const settingsEndpoint = new RegExp(
+		'^/google-site-kit/v1/modules/reader-revenue-manager/data/settings'
+	);
+
 	beforeAll( () => {
 		API.setUsingCache( false );
 	} );
 
 	beforeEach( () => {
+		enabledFeatures.add( 'rrmModule' ); // Enable RRM module to get its features.
 		registry = createTestRegistry();
+		provideUserInfo( registry );
 	} );
 
-	describe( 'selectors', () => {
-		describe( 'getPublications', () => {
-			it( 'should use a resolver to make a network request', async () => {
-				fetchMock.get( publicationsEndpoint, {
-					body: fixtures.publications,
-					status: 200,
-				} );
+	describe( 'actions', () => {
+		beforeEach( () => {
+			// Make sure the RRM module is active and connected.
+			const extraData = [
+				{
+					slug: MODULE_SLUG,
+					active: true,
+					connected: true,
+				},
+			];
+			provideModules( registry, extraData );
+			provideModuleRegistrations( registry, extraData );
+		} );
 
-				const initialPublications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( initialPublications ).toBeUndefined();
+		describe( 'syncPublicationOnboardingState', () => {
+			it( 'should return undefined when no publication ID is present', async () => {
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetPublications( fixtures.publications );
 
-				await untilResolved(
-					registry,
-					MODULES_READER_REVENUE_MANAGER
-				).getPublications();
-				expect( fetchMock ).toHaveFetched( publicationsEndpoint );
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetSettings( {} );
 
-				const publications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( fetchMock ).toHaveFetchedTimes( 1 );
-				expect( publications ).toEqual( fixtures.publications );
-				expect( publications ).toHaveLength(
-					fixtures.publications.length
-				);
+				const syncStatus = await registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.syncPublicationOnboardingState();
+
+				expect( syncStatus ).toBeUndefined();
 			} );
 
-			it( 'should not make a network request if publications are already present', async () => {
+			it( 'should update the settings and call the saveSettings endpoint', async () => {
 				registry
 					.dispatch( MODULES_READER_REVENUE_MANAGER )
 					.receiveGetPublications( fixtures.publications );
 
-				const publications = registry
+				const publication = fixtures.publications[ 0 ];
+
+				const settings = {
+					publicationID: publication.publicationId,
+					publicationOnboardingState:
+						PUBLICATION_ONBOARDING_STATES.PENDING_VERIFICATION,
+					publicationOnboardingStateLastSyncedAtMs: 0,
+				};
+
+				fetchMock.post( settingsEndpoint, {
+					body: {
+						...settings,
+						publicationOnboardingState:
+							PUBLICATION_ONBOARDING_STATES.UNSPECIFIED,
+						publicationOnboardingStateLastSyncedAtMs: Date.now(),
```

Comment:
If we're using a mocked `Date.now()` value, this can be updated to reflect it:

```suggestion
						publicationOnboardingStateLastSyncedAtMs: mockNow,
```